### PR TITLE
Return error when the data in bucket auth secret is empty

### PR DESCRIPTION
**Pull request by semihbkgr**
It should return an error when the auth data (secret fields) are not provided in the given secret. 
Otherwise this function will return nil secret map. In the rest of the operations, we consider that the user didn't give any secret, and we will try Workload Identity authentication.

### CODE REVIEW

File: internal/bucket/bucket.go [hunk 1]
```diff
@@ -89,6 +89,11 @@ func SecretData(ctx context.Context, sn string) (map[string][]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if len(secret.Data) == 0 {
+		return nil, fmt.Errorf("data is empty in the bucket authentication secret: %s", secret.Name)
```
Comment (kutluhanmetin) [inline internal/bucket/bucket.go:94]:
```suggestion
		return nil, fmt.Errorf("the data in the bucket authentication secret is empty: %s", secret.Name)
```

Comment (semihbkgr) [reply internal/bucket/bucket.go:94]: besides this, we can consider to test this logic by mocking k8s client

**Review by kutluhanmetin — APPROVED** [review]

**Review by SeriyBg — COMMENTED** [review]
I know we didn't do it much in the past, but for me it makes sense to add an unit test for this case 😄

File: internal/k8sutil/client.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package k8sutil
```
Comment (SeriyBg) [inline internal/k8sutil/client.go:1]: `util` in namig is considered an anti-pattern in go. Can we rename it to `k8s`?

File: internal/k8sutil/namespace.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package k8sutil
+
+import (
+	"os"
+	"strings"
+)
+
+func Namespace() (string, error) {
```
Comment (SeriyBg) [inline internal/k8sutil/namespace.go:8]: Can't we have these two functions in one file?

File: internal/bucket/bucket_test.go [hunk 4]
```diff
@@ -171,3 +174,71 @@ func TestDownloadFile(t *testing.T) {
...
+
+func fakeSecretReader(name string, data map[string][]byte) SecretReader {
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "default",
+		},
+		Data: data,
+	}
+	c := fake.NewSimpleClientset(&secret)
```
Comment (SeriyBg) [inline internal/bucket/bucket_test.go:242]:
```suggestion
	secret := &corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: "default",
		},
		Data: data,
	}
	c := fake.NewSimpleClientset(secret)
```

File: internal/bucket/bucket.go [hunk 5]
```diff
@@ -68,44 +68,39 @@ func OpenBucket(ctx context.Context, bucketURL string, secretData map[string][]b
...
+type SecretReader struct {
+	clientcorev1.SecretInterface
+}
 
-	config, err := rest.InClusterConfig()
+func NewSecretReader() (SecretReader, error) {
+	c, err := k8sutil.Client()
 	if err != nil {
-		return nil, err
+		return SecretReader{}, err
```
Comment (SeriyBg) [inline internal/bucket/bucket.go:78]: Can we instead return `nil` and change the return type to `*SecretReader`? It is better not to create an empty object that won't be used

File: internal/bucket/bucket_test.go [hunk 6]
```diff
@@ -171,3 +174,71 @@ func TestDownloadFile(t *testing.T) {
...
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			c := fake.NewSimpleClientset()
+			sr := SecretReader{SecretInterface: c.CoreV1().Secrets("")}
+			if test.secretName != "" && test.data != nil {
+				sr = fakeSecretReader(test.secretName, test.data)
+			}
+			data, err := sr.SecretData(context.Background(), test.secretName)
+			if test.errMsg != "" {
```
Comment (SeriyBg) [inline internal/bucket/bucket_test.go:223]: It's a small nit, but I always prefer two separate tests for such cases. It makes it harder to maintain when the assertions in the tests are based on some conditions. It's better to have a test for case where error is expected, and a separate one when the error is not expected

File: internal/bucket/bucket.go (hunk 5 above)
Comment (semihbkgr) [reply internal/bucket/bucket.go:78]: Yes we can. My thought was to return object itself as its all fields are pointer.

File: internal/bucket/bucket_test.go (hunk 6 above)
Comment (semihbkgr) [reply internal/bucket/bucket_test.go:223]: tbh I don't want to touch it for now. We can refactor such kind of things after adding all unit tests.

File: internal/bucket/bucket.go (hunk 5 above)
Comment (SeriyBg) [reply internal/bucket/bucket.go:78]: Yes, but it's not common to return an empty object that won't be used. Besides, it's recommended to use pointer unless there is a reason not to use one

File: internal/bucket/bucket_test.go (hunk 6 above)
Comment (SeriyBg) [reply internal/bucket/bucket_test.go:223]: We won't add a lot of unit tests for now. This change will only create one more case will similar logic, just a sparate assertions

**Review by SeriyBg — APPROVED** [review]

**Pull request merged**